dotCMS/src/main/java/com/dotmarketing/common/db/ForeignKey.java

```diff
@@ -0,0 +1,95 @@
+package com.dotmarketing.common.db;
+
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+public class ForeignKey {
+
+    private final String primaryKeyTableName;
+    private final Set<String> primaryKeyColumnNames = new LinkedHashSet<>();
+    private final String foreignKeyTableName;
+    private final Set<String> foreignKeyColumnNames = new LinkedHashSet<>();
+    private final String foreignKeyName;
+
+    public ForeignKey(String primaryKeyTableName, String foreignKeyTableName, String foreignKeyName) {
+        this.primaryKeyTableName = primaryKeyTableName;
+        this.foreignKeyTableName = foreignKeyTableName;
+        this.foreignKeyName = foreignKeyName;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 5, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1142)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This class overrides "equals()" and should therefore also override "hashCode()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1206)

Reply:
DONE